test/deployframework/helpers.go

```diff
@@ -289,3 +284,291 @@ func cleanupLocalCmds(logger logrus.FieldLogger, commands ...exec.Cmd) error {
+	// and that Pod is reporting a Ready stauts, or stop when the timeout period is reached.
+	err := wait.Poll(3*time.Second, 1*time.Minute, func() (done bool, err error) {
+		pods, err := client.CoreV1().Pods(namespace).List(meta.ListOptions{
+			LabelSelector: fmt.Sprintf("olm.catalogSource=%s", packageName),
+		})
+		if err != nil {
+			return false, err
+		}
+		if len(pods.Items) != 1 {
+			return false, nil
+		}
+
+		for _, pod := range pods.Items {
+			podIsReady, _ := checkPodStatus(pod)
+			if !podIsReady {
+				logger.Infof("Waiting for the %s Pod to become Ready", pod.Name)
+				return false, nil
+			}
+		}
+
+		return true, nil
+	})
+	if err != nil {
+		return fmt.Errorf("failed to wait for the %s catalogsource Pod to become ready: %v", packageName, err)
+	}
+
+	return nil
+}
+
+// CreateUpgradeConfigMap is a helper function responsible for creating a ConfigMap
+// that contains the current version of the repositories' CRDs, CSV and metering-ocp package
+// which OLM can then consume through a CatalogSource. In order to create this ConfigMap,
+// we execute a bash script that handles the heavy-lifting, overriding any of the environment
+// variables that the script uses, to match our current deployment context.
+func CreateUpgradeConfigMap(logger logrus.FieldLogger, name, namespace, scriptPath string) error {
+	/*
+		Check if we are running in CI by getting the value of the
+		IMAGE_FORMAT environment variable that CI builds and exposes
+		for our job. If this value is non-empty, then the "update
+		configmap" script will override the containerImage field in the CSV.
+		Else, the containerImage will use the default origin images.
+
+		More information:
+		https://github.com/openshift/ci-tools/blob/master/TEMPLATES.md#image_format
+	*/
+	imageOverride := os.Getenv("IMAGE_FORMAT")
+	if imageOverride != "" {
+		imageOverride = strings.Replace(imageOverride, "${component}", "metering-ansible-operator", 1)
+	}
+
+	envVarArr := []string{
+		"IMAGE_OVERRIDE=" + imageOverride,
+		"NAMESPACE=" + namespace,
+		"NAME=" + name,
+	}
+
+	// build up the path to the ./hack/@scriptPath and stat that path,
+	// verifying it exists before running that bash script
+	relPath := filepath.Join(scriptPath, createUpgradeConfigMapScriptName)
+	createConfigMapScript, err := filepath.Abs(relPath)
+	if err != nil {
+		return fmt.Errorf("failed to get the absolute path for the '%s' path: %v", relPath, err)
+	}
+	_, err = os.Stat(createConfigMapScript)
+	if err != nil {
+		return fmt.Errorf("failed to stat the '%s' path: %v", createConfigMapScript, err)
+	}
+
+	cmd := exec.Command(createConfigMapScript)
+	cmd.Env = append(os.Environ(), envVarArr...)
+	stderr, _ := cmd.StderrPipe()
+
+	err = cmd.Start()
+	if err != nil {
+		return fmt.Errorf("failed to start running the %s script", createConfigMapScript)
+	}
+
+	scanner := bufio.NewScanner(stderr)
+	scanner.Split(bufio.ScanWords)
+	for scanner.Scan() {
+		m := scanner.Text()
+		fmt.Println(m)
+	}
+
+	err = cmd.Wait()
```

Comment:
This shouldn't really wait forever should it?

Reply:
This shouldn't be a problem here and updating the implementation for this (and the other functions/methods in the testing suite) to adding and managing go routines isn't something I want to explore in this current PR. I can add a TODO comment so we can look more into adding a timeout function.

Reply:
TODO sounds like way to go.